RecoJets/JetProducers/python/PileupJetID_cfi.py

```diff
@@ -14,8 +14,9 @@
 _chsalgos_94x = cms.VPSet(full_94x_chs,cutbased)
 _chsalgos_102x = cms.VPSet(full_102x_chs,cutbased)
 _chsalgos_106X_UL17 = cms.VPSet(full_106x_UL17_chs,cutbased)
+_chsalgos_106X_UL18 = cms.VPSet(full_106x_UL18_chs,cutbased)
 
-_stdalgos    = _chsalgos_106X_UL17
+_stdalgos    = _chsalgos_106X_UL18
```

Comment:
is this new default applicable to all inputs?
I'd naively assume that the simulations or data from 2017 or 2018 should get different values.

Perhaps reco done from scratch should still use `_chsalgos_106X_UL18`, but the available AODs perhaps better use the matching year training PU IDs.

The era-driven mods will need to be applied below on pileupJetId object  like e.g. `(run2_miniAOD_UL & run2_jme_2017).toModify(pileupJetId, algos = _chsalgos_106X_UL17)`

Note that we also support running minis for 94X and 80X AODs,
are they correctly set to use `_chsalgos_106X_*`? or should they better pick up the older variant?
It's not obvious what supersedes, the kind of training or the kind of data.

Reply:
Yes respective years should get their trained PUID.

From previous conversations https://github.com/cms-sw/cmssw/pull/28827#issuecomment-581347782 it seems `_stdalgos` should point to the latest.

Reply:
Yes, the default is typically selected to cover nearest future case as well. So, the current choice for it is OK.

However, the modules defined below and using `_stdalgos` should be updated/modified appropriately to pick up the right variant for a given set of inputs; the UL17 case has a straightforward choice in https://github.com/cms-sw/cmssw/pull/32418/files#r545504015; I'm not sure what to suggest for UL16 and also for the older 94X and 80X inputs.

Reply:
Hi @slava77 can you give some hint here, what it needs to do is pick respective algo for respective year, 94X 2016,2017,2018 and UL 2016,2017,2018.
By default what it is picking up the latest?
Should I import all 6 (94X 3years, UL  3years) configuration and do `.toModify` ?

Reply:
at least for miniAOD, 
please elaborate what's the correct mapping for a set of version input and year pairs where input is among (80X, 94X, UL), and year out of (2016, 2017, 2018)

Reply:
I can only say it for miniAOD,
miniAOD version | Run2 Year | Algo
------------------- | ---------- | ----------
94X | 2016 | _chsalgos_81x
94X | 2017 | _chsalgos_94x
94X | 2018 | _chsalgos_102x
UL | 2017 | _chsalgos_106X_UL17
UL | 2018 | _chsalgos_106X_UL18

and UL 2016, when it's complete.

Reply:
- I'm surprised to see 2018 in the 94X list (2016 remini and 2017 [re]reco-based mini are OK).
- what about the 80X 2016 legacy miniAOD? should it be the `_chsalgos_81x`?

Reply:
yes, sorry 2018 ~94X~ is 102X (made a mistake here). and 2016 80X should be same as 2016 94X.

Reply:
OK, so, similar to what we have already in 10_6_X https://github.com/cms-sw/cmssw/blob/CMSSW_10_6_X/RecoJets/JetProducers/python/PileupJetID_cfi.py#L36
we will need to add modifications in line with the table above

```python
run2_miniAOD_80XLegacy.toModify(pileupJetId, algos = _chsalgos_81x)
run2_miniAOD_94XFall17.toModify(pileupJetId, algos = _chsalgos_94x)
(run2_miniAOD_UL & run2_jme_2017).toModify(pileupJetId, algos = _chsalgos_106X_UL17)
```

among the supported re-miniAOD workflows we do not have anything using 102X AOD inputs for 2018 processing.
So, the list above should cover all that we have for re-miniAOD.

Still, what remains is to understand/confirm which settings should be used for reconstruction from scratch/RAW.
Currently, everything made from scratch/RAW will get the new default `_chsalgos_106X_UL18`. Is this appropriate? My guess is that `_chsalgos_106X_UL17` and a future `_chsalgos_106X_UL16` may be better for the corresponding year(s). In that case the last line above will be (by exclusion)
```python
 ( (~run2_miniAOD_94XFall17) & run2_jme_2017).toModify(pileupJetId, algos = _chsalgos_106X_UL17)
```

Reply:
> Still, what remains is to understand/confirm which settings should be used for reconstruction from scratch/RAW.
> Currently, everything made from scratch/RAW will get the new default `_chsalgos_106X_UL18`. Is this appropriate? My guess is that `_chsalgos_106X_UL17` and a future `_chsalgos_106X_UL16` may be better for the corresponding year(s). In that case the last line above will be (by exclusion)
> 

@alefisico @camclean can you comment on this?

Reply:
Hi @slava77, I tested your suggestion, it is picking up correct algos for `80XLegacy` and `94XFall17`. But for `U17` it is still picking up `UL18` with, `(run2_miniAOD_UL & run2_jme_2017).toModify(pileupJetId, algos = _chsalgos_106X_UL17)`

I tested by dumping python config, for example ( I got it from existing samples production command)

```bash
cmsDriver.py HIG-RunIISummer19UL17MiniAOD-00456_1_cfg.py  --dump_python --eventcontent MINIAODSIM --datatier MINIAODSIM --conditions 106X_mc2017_realistic_v6 --step PAT --geometry DB:Extended --era Run2_2017 --no_exec --mc
```

if you have other test to do locally before I make a commit, I can also do that.

Reply:
the `cmsDriver.py ...  --era Run2_2017 ...`  above is appropriate for running miniAOD from scratch and would not match the re-miniAOD UL `(run2_miniAOD_UL & run2_jme_2017)`

For re-miniAOD UL there should be `--procModifiers run2_miniAOD_UL` (or `run2_miniAOD_UL_preSummer20` depending on the campaign for the input sample)

Reply:
an example re-miniAOD workflow configuration using 2017 UL inputs can be obtained from `runTheMatrix.py -l 136.83111 -n -e`

Reply:
thanks. it worked.
I pushed the commit with first suggestion.